presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadata.java

```diff
@@ -2194,16 +2232,32 @@ public void commit()
         metastore.commit();
     }
 
-    public static boolean isPartitionsSystemTable(SchemaTableName tableName)
+    public static Optional<SchemaTableName> getSourceTableNameFromSystemTable(SchemaTableName tableName)
     {
-        return tableName.getTableName().endsWith(PARTITIONS_TABLE_SUFFIX) && tableName.getTableName().length() > PARTITIONS_TABLE_SUFFIX.length();
+        return Stream.of(PARTITIONS_TABLE_HANDLER, PROPERTIES_TABLE_HANDLER)
+                .filter(handler -> handler.matches(tableName))
+                .map(handler -> handler.getSourceTableName(tableName))
+                .findAny();
     }
 
-    public static SchemaTableName getSourceTableNameForPartitionsTable(SchemaTableName tableName)
+    private interface SystemTableHandler
```

Comment:
That's a good idea. It could look like
```java
private enum SystemTableHandler
{
    PARTITIONS, PROPERTIES;

    private final String suffix;

    SystemTableHandler()
    {
        this.suffix = "$" + name().toLowerCase(ENGLISH);
    }

    public boolean matches(...)

    SchemaTableName getSourceTableName(...)
}